Added possibility to upload device_info.json

alexandru-nicolescu opened the pull request:
+ HAPairController.js - controller for HAPair model
+ HAPair.js - HAPair model file
+ add_ha_pair.css - styling the Add HA Pair view
+ add_ha_pair.ctrl.js - functionality for uploading the device_info.json file
+ add_ha_pair.tml.html - template for Add HA Pair view
- Modified nav-bar.tmpl.html to include the Add HA Pair option
- Modified frontend/app.js to create a route for Add HA Pair view

Commit:
Added possibility to upload device_info.json

+ HAPairController.js - controller for HAPair model
+ HAPair.js - HAPair model file
+ add_ha_pair.css - styling the Add HA Pair view
+ add_ha_pair.ctrl.js - functionality for uploading the device_info.json file
+ add_ha_pair.tml.html - template for Add HA Pair view
- Modified nav-bar.tmpl.html to include the Add HA Pair option
- Modified frontend/app.js to create a route for Add HA Pair view

cristiannaziru commented on frontend/app/components/ha_pair/add_ha_pair.ctrl.js:
```diff
@@ -0,0 +1,60 @@
+        }
+
+        function add_ha_pair(ha_pair) {
+            $http.post('/api/hapair', ha_pair).then(response => {
```
Hey @alexandru-nicolescu ! Let's create a Service for the http requests in case we'll need them in other places, we can simply reuse that.

Services are singletons and don't need to be instantiate, just simply injected ($http is a service for example) and then used: HAPairService.add_pair(ha_pair). etc.. 

This will also help you understand how to pass callbacks/promises from one file to another. 
Here's official docs on services: https://docs.angularjs.org/guide/services#using-a-service 

Let me know if you need help.

cristiannaziru commented on frontend/temp/index.html:
```diff
@@ -11,13 +11,14 @@
     <!-- inject:css -->
```
The files from /temp folder shouldn't be committed / modified as they are generated when you run gulp. I'm sorry for the trouble, I shouldn't have committed it here, and I should've added it to .gitignore. Feel free to do it yourself (and u can also push changes of removing the temp folder from this repo).

cristiannaziru commented on frontend/temp/app.js:
```diff
@@ -25,5 +25,9 @@ app.config(function($routeProvider) {
     templateUrl : '/components/upgrade/upgrade.tmpl.html',
```
Same as the below comment. (We don't need /temp folder)

Commit:
Untrack temp/app.js and temp/index.html

Commit:
Untrack bower_components

Commit:
Added HA Pair service

+ ha-pair.js - functionality for adding a HA Pair
- Modified add_ha_pair.ctrl.js to use the HA Pair service

cristiannaziru commented on frontend/app/services/ha_pair/ha_pair.js:
```diff
@@ -0,0 +1,10 @@
+    HAPairService.$inject = ['$http'];
+    function HAPairService($http) {
+        this.add_ha_pair = function (ha_pair) {
+            return $http.post('/api/hapair', ha_pair);
```
Nice! The takeaway from here is that the $http service returns a promise, you return it from this.add_ha_pair function and then you resolve it in your controller.

cristiannaziru reviewed (approved):
Great job 👍
